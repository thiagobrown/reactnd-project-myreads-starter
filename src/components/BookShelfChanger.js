import React from 'react'
import PropTypes from 'prop-types'

const BookShelfChanger = ({ book, onUpdateBook }) => {
  return (
    <div className="book-shelf-changer">
      <select value={book.shelf}
        onChange={(e) => onUpdateBook(book, e.target.value)}>
        <option value="" disabled>Move to...</option>
        <option value="currentlyReading">Currently Reading</option>
        <option value="wantToRead">Want to Read</option>
        <option value="read">Read</option>
        <option value="none">None</option>
      </select>
    </div>
  )
}

BookShelfChanger.propTypes = {
  book: PropTypes.object.isRequired,
  onUpdateBook: PropTypes.func.isRequired
}

export default BookShelfChanger;
