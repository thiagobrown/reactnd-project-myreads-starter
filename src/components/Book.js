import React from 'react'
import PropTypes from 'prop-types'

import BookShelfChanger from './BookShelfChanger'

const Book = ({ book, onUpdateBook }) => {
    return (
        <li>
            <div className="book">
                <div className="book-top">
                    <div className="book-cover" style={{ width: 128, height: 193, backgroundImage: `url(${book.imageLinks.thumbnail})` }}>
                    </div>
                    <BookShelfChanger book={book} onUpdateBook={onUpdateBook} />
                </div>
                <div className="book-title">{book.title}</div>
                <div className="book-authors">{book.authors && book.authors.length > 1 ? book.authors.join(", ") : book.authors}</div>
            </div>
        </li>
    )
}

Book.propTypes = {
    book: PropTypes.object.isRequired,
    onUpdateBook: PropTypes.func.isRequired
}

export default Book;
