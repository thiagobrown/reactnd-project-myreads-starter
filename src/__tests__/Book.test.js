import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import Book from './../components/Book'
import BookShelfChanger from './../components/BookShelfChanger'

describe('Testing Book Component', () => {
  let props = {
    book: {
      id: 1,
      title: 'Book One',
      authors: ['Authors One'],
      imageLinks: { thumbnail: 'url/book/one' },
      shelf: 'read'
    },
    onUpdateBook: jest.fn()
  }

  const wrapper = shallow(<Book {...props} />);

  it('render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('verify exists values passed as props', () => {
    expect(
      wrapper.find('div.book > div.book-top > div.book-cover')
      .props().style.backgroundImage)
      .toBe(`url(${props.book.imageLinks.thumbnail})`);
    
    expect(
      wrapper.find('div.book > div.book-title').text())
      .toBe(props.book.title);
    
    expect(
      wrapper.find('div.book > div.book-authors').text())
      .toBe(props.book.authors.join(', '));
  });

  it('verify exists BookShelfChanger', () => {
    expect(wrapper.find(BookShelfChanger)).toBeTruthy();
  })
});



