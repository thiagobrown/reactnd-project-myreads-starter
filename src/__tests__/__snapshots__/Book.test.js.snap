exports[`Testing Book Component render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Book
    book={
        Object {
            "authors": Array [
              "Authors One",
            ],
            "id": 1,
            "imageLinks": Object {
              "thumbnail": "url/book/one",
            },
            "shelf": "read",
            "title": "Book One",
          }
    }
    onUpdateBook={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="book">
        <div
                className="book-top">
                <div
                        className="book-cover"
                        style={
                                Object {
                                        "backgroundImage": "url(url/book/one)",
                                        "height": 193,
                                        "width": 128,
                                      }
                        } />
                <BookShelfChanger
                        book={
                                Object {
                                        "authors": Array [
                                          "Authors One",
                                        ],
                                        "id": 1,
                                        "imageLinks": Object {
                                          "thumbnail": "url/book/one",
                                        },
                                        "shelf": "read",
                                        "title": "Book One",
                                      }
                        }
                        onUpdateBook={[Function]} />
        </div>
        <div
                className="book-title">
                Book One
        </div>
        <div
                className="book-authors">
                Authors One
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="book-top">
            <div
                        className="book-cover"
                        style={
                                    Object {
                                                "backgroundImage": "url(url/book/one)",
                                                "height": 193,
                                                "width": 128,
                                              }
                        } />
            <BookShelfChanger
                        book={
                                    Object {
                                                "authors": Array [
                                                  "Authors One",
                                                ],
                                                "id": 1,
                                                "imageLinks": Object {
                                                  "thumbnail": "url/book/one",
                                                },
                                                "shelf": "read",
                                                "title": "Book One",
                                              }
                        }
                        onUpdateBook={[Function]} />
</div>,
          <div
            className="book-title">
            Book One
</div>,
          <div
            className="book-authors">
            Authors One
</div>,
        ],
        "className": "book",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="book-cover"
                style={
                                Object {
                                                "backgroundImage": "url(url/book/one)",
                                                "height": 193,
                                                "width": 128,
                                              }
                } />,
              <BookShelfChanger
                book={
                                Object {
                                                "authors": Array [
                                                  "Authors One",
                                                ],
                                                "id": 1,
                                                "imageLinks": Object {
                                                  "thumbnail": "url/book/one",
                                                },
                                                "shelf": "read",
                                                "title": "Book One",
                                              }
                }
                onUpdateBook={[Function]} />,
            ],
            "className": "book-top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "book-cover",
                "style": Object {
                  "backgroundImage": "url(url/book/one)",
                  "height": 193,
                  "width": 128,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "book": Object {
                  "authors": Array [
                    "Authors One",
                  ],
                  "id": 1,
                  "imageLinks": Object {
                    "thumbnail": "url/book/one",
                  },
                  "shelf": "read",
                  "title": "Book One",
                },
                "onUpdateBook": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Book One",
            "className": "book-title",
          },
          "ref": null,
          "rendered": "Book One",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Authors One",
            ],
            "className": "book-authors",
          },
          "ref": null,
          "rendered": Array [
            "Authors One",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="book">
          <div
                    className="book-top">
                    <div
                              className="book-cover"
                              style={
                                        Object {
                                                  "backgroundImage": "url(url/book/one)",
                                                  "height": 193,
                                                  "width": 128,
                                                }
                              } />
                    <BookShelfChanger
                              book={
                                        Object {
                                                  "authors": Array [
                                                    "Authors One",
                                                  ],
                                                  "id": 1,
                                                  "imageLinks": Object {
                                                    "thumbnail": "url/book/one",
                                                  },
                                                  "shelf": "read",
                                                  "title": "Book One",
                                                }
                              }
                              onUpdateBook={[Function]} />
          </div>
          <div
                    className="book-title">
                    Book One
          </div>
          <div
                    className="book-authors">
                    Authors One
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="book-top">
              <div
                            className="book-cover"
                            style={
                                          Object {
                                                        "backgroundImage": "url(url/book/one)",
                                                        "height": 193,
                                                        "width": 128,
                                                      }
                            } />
              <BookShelfChanger
                            book={
                                          Object {
                                                        "authors": Array [
                                                          "Authors One",
                                                        ],
                                                        "id": 1,
                                                        "imageLinks": Object {
                                                          "thumbnail": "url/book/one",
                                                        },
                                                        "shelf": "read",
                                                        "title": "Book One",
                                                      }
                            }
                            onUpdateBook={[Function]} />
</div>,
            <div
              className="book-title">
              Book One
</div>,
            <div
              className="book-authors">
              Authors One
</div>,
          ],
          "className": "book",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="book-cover"
                  style={
                                    Object {
                                                      "backgroundImage": "url(url/book/one)",
                                                      "height": 193,
                                                      "width": 128,
                                                    }
                  } />,
                <BookShelfChanger
                  book={
                                    Object {
                                                      "authors": Array [
                                                        "Authors One",
                                                      ],
                                                      "id": 1,
                                                      "imageLinks": Object {
                                                        "thumbnail": "url/book/one",
                                                      },
                                                      "shelf": "read",
                                                      "title": "Book One",
                                                    }
                  }
                  onUpdateBook={[Function]} />,
              ],
              "className": "book-top",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "book-cover",
                  "style": Object {
                    "backgroundImage": "url(url/book/one)",
                    "height": 193,
                    "width": 128,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "book": Object {
                    "authors": Array [
                      "Authors One",
                    ],
                    "id": 1,
                    "imageLinks": Object {
                      "thumbnail": "url/book/one",
                    },
                    "shelf": "read",
                    "title": "Book One",
                  },
                  "onUpdateBook": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Book One",
              "className": "book-title",
            },
            "ref": null,
            "rendered": "Book One",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Authors One",
              ],
              "className": "book-authors",
            },
            "ref": null,
            "rendered": Array [
              "Authors One",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
